tensorrt_llm/_torch/models/checkpoints/weight_mapper_interface.py

```diff
@@ -0,0 +1,174 @@
+from abc import ABC, abstractmethod
+from functools import wraps
+from typing import Callable, List, Union
+
+from torch import nn
+
+from tensorrt_llm._torch.model_config import ModelConfig, TConfig
+from tensorrt_llm._torch.models.modeling_utils import DecoderModelForCausalLM
+
+
+def guard_all_methods(cls):
```

Comment:
I think i understand where youre coming from with this, but im wondering if there may be a cleaner way. Im sure you dont like it either.

Not to get too philosophical, but this looks like a half-baked `Builder` pattern type of thing. Lets brainstorm 15 mins F2F if theres some hack to make this a bit prettier (definitely not sure we'll find one...).